api/src/main/java/io/druid/data/input/InputRow.java

```diff
@@ -28,8 +30,8 @@
  * implement "schema-less" data ingestion that allows the system to add new dimensions as they appear.
  *
  */
-public interface
-    InputRow extends Row
+@PublicApi
```

Comment:
shouldn't this be an extension point instead ?  InputRowParser is an extension point and parse(..) should be free to return a concrete implementation that it wants.